### PR TITLE
Added buttons to enable admins to edit task

### LINKED ISSUE #222: Enable edit mode for `super user`
Enable edit mode for `super user || admin user` to re-route to `/?edit=true` on click of button

## Requirements
- [x] Show a pencil icon on the ~~top-right of the screen~~ bottom-left of the cards in tasks
- [x] Keep it hidden by default
- [x] Make it visible on the long-press (300ms) of the Option/ALT
- [x] #246

Reach out to @harshith-venkatesh  for more information
image

### PR DESCRIPTION
**Pull request by Kshashank99**
Closes #222 
### What is the change?

Added buttons to allow superuser|| admin to re-route to `?edit=true`  and `?dev=true`

### Is it a bug?

- No

### \*Dev Tested?

- [x] Yes
- [ ] No

#### Platforms

- [x] Web

#### Browsers

- [x] Chrome
- [x] Safari
- [x] Firefox

### Before / After Change Screenshots

> For visual or interaction changes. Can be video / screenshot.

https://user-images.githubusercontent.com/64394213/153803603-3651b11b-c47e-4ad6-bc32-26dbe3c1fde4.mp4

### CODE REVIEW

File: src/components/Layout/Layout.module.scss [hunk 1]
```diff
@@ -30,4 +31,17 @@
 
 .active {
   color: #E30464;
-}
\ No newline at end of file
+}
+.edit{
+  position: relative;
+  right: 0;
```
Comment (DeRaowl) [inline src/components/Layout/Layout.module.scss:37]: As right is used below, can you remove this line

File: src/components/Layout/Layout.module.scss [hunk 2]
```diff
@@ -30,4 +31,17 @@
...
+  cursor: pointer;
+  // padding-right:25%;
+  float:right;
+  right:250px;
```
Comment (DeRaowl) [inline src/components/Layout/Layout.module.scss:41]: Can you use relative units?

File: src/components/Layout/Layout.module.scss [hunk 3]
```diff
@@ -30,4 +31,17 @@
...
+
+}
+// .links{
```
Comment (DeRaowl) [inline src/components/Layout/Layout.module.scss:44]: Commented Code can be removed.

File: src/components/Layout/index.tsx [hunk 4]
```diff
@@ -18,14 +20,40 @@ const navBarContent = (title: string, refUrl: string, isActive: boolean = false)
...
+  }, []);
+
   const router = useRouter();
 
   // Dev feature toggle
   const { query } = router;
-  const dev = !!query.dev;
-
+  let dev = !!query.dev;
+  let elementRef  = useRef();
```
Comment (DeRaowl) [inline src/components/Layout/index.tsx:55]: useRef created but never used

File: src/components/Layout/index.tsx [hunk 5]
```diff
@@ -1,10 +1,12 @@
-import { FC, ReactNode } from 'react';
+import { FC, ReactNode, useEffect, useRef, useState } from 'react';
```
Comment (DeRaowl) [inline src/components/Layout/index.tsx:1]: useRef imported and never used

**Review by DeRaowl — CHANGES_REQUESTED** [review]

File: src/components/Layout/Layout.module.scss [hunk 6]
```diff
@@ -30,4 +31,17 @@
...
 .active {
   color: #E30464;
-}
\ No newline at end of file
+}
+.edit{
+  position: relative;
+  right: 0;
+  cursor: pointer;
+  // padding-right:25%;
```
Comment (Pavangbhat) [inline src/components/Layout/Layout.module.scss:39]: please remove comments in the code

File: src/components/Layout/index.tsx [hunk 7]
```diff
@@ -18,14 +20,40 @@ const navBarContent = (title: string, refUrl: string, isActive: boolean = false)
     </Link>
   );
 };
-
+const { SUCCESS, ERROR } = ToastTypes;
+const SELF_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/users/self`;
```
Comment (Pavangbhat) [inline src/components/Layout/index.tsx:24]: Please make use of constants from here 👇 
https://github.com/Real-Dev-Squad/website-status/blob/develop/src/components/constants/url.js

File: src/components/Layout/index.tsx [hunk 8]
```diff
@@ -18,14 +20,40 @@ const navBarContent = (title: string, refUrl: string, isActive: boolean = false)
     </Link>
   );
 };
-
+const { SUCCESS, ERROR } = ToastTypes;
```
Comment (Pavangbhat) [inline src/components/Layout/index.tsx:23]: Please move this line to the end of the import statements

File: src/components/Layout/index.tsx [hunk 9]
```diff
@@ -51,7 +80,25 @@ const Layout: FC<Props> = ({ children }) => {
               </>
             )
           }
+          {IsUserAuthorized && <Link href={{ pathname: '/', query: { dev: 'true' } }} passHref><Image
+                height= "25px"
+                width="25px"
+                src="/add.png"
```
Comment (Pavangbhat) [inline src/components/Layout/index.tsx:86]: I think we can remove **/** here

File: src/components/Layout/index.tsx [hunk 10]
```diff
@@ -51,7 +80,25 @@ const Layout: FC<Props> = ({ children }) => {
...
+                alt="add"
+            /></Link>}  
         </div>
+        {IsUserAuthorized && <div className={styles.edit} >
+          <Link href={{ pathname: '/', query: { edit: 'true' } }} passHref>
+            <Image
+                className={styles.edit}
+                height= "25px"
+                width="25px"
+                src="/pencil.png"
```
Comment (Pavangbhat) [inline src/components/Layout/index.tsx:96]: I think we can remove **/** here

**Review by Pavangbhat — CHANGES_REQUESTED** [review]
I have commented few suggestions, please have a look

File: src/components/Layout/Layout.module.scss [hunk 11]
```diff
@@ -30,4 +31,17 @@
 
 .active {
   color: #E30464;
```
Comment (Neha) [inline src/components/Layout/Layout.module.scss:33]: This should move to the variable file

File: src/components/Layout/Layout.module.scss [hunk 12]
```diff
@@ -30,4 +31,17 @@
...
-}
\ No newline at end of file
+}
+.edit{
+  position: relative;
+  right: 0;
+  cursor: pointer;
+  // padding-right:25%;
+  float:right;
```
Comment (Neha) [inline src/components/Layout/Layout.module.scss:40]: Do not use `FLOAT`

File: src/components/Layout/Layout.module.scss (hunk 2 above)
Comment (Neha) [reply src/components/Layout/Layout.module.scss:41]: No, relative units should not be used.

File: src/components/Layout/index.tsx [hunk 13]
```diff
@@ -18,14 +20,40 @@ const navBarContent = (title: string, refUrl: string, isActive: boolean = false)
     </Link>
   );
 };
-
+const { SUCCESS, ERROR } = ToastTypes;
+const SELF_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/users/self`;
 const Layout: FC<Props> = ({ children }) => {
+  const [IsUserAuthorized, setIsUserAuthorized] = useState(false);
+
+  useEffect(() => {
```
Comment (Neha) [inline src/components/Layout/index.tsx:28]: Move this to a separate function (util). Keep the component only for the single task it is responsible for

File: src/components/Layout/index.tsx [hunk 14]
```diff
@@ -42,6 +70,7 @@ const Layout: FC<Props> = ({ children }) => {
           {navBarContent('Stale PRs', '/stale-pr', router.pathname === '/stale-pr')}
```
Comment (Neha) [inline src/components/Layout/index.tsx:70]: Move hard-coded content such as _Stale PRs_ to the `constant` file

File: src/components/Layout/index.tsx [hunk 15]
```diff
@@ -42,6 +70,7 @@ const Layout: FC<Props> = ({ children }) => {
...
           |
           {navBarContent('Idle Members', '/idle-members', router.pathname === '/idle-members')}
+
```
Comment (Neha) [inline src/components/Layout/index.tsx:73]: `Routing` should be move to the separate component/file

File: src/components/Layout/index.tsx [hunk 16]
```diff
@@ -51,7 +80,25 @@ const Layout: FC<Props> = ({ children }) => {
               </>
             )
           }
+          {IsUserAuthorized && <Link href={{ pathname: '/', query: { dev: 'true' } }} passHref><Image
+                height= "25px"
```
Comment (Neha) [inline src/components/Layout/index.tsx:84]: Move this to the `constant` file

**Review by Neha — CHANGES_REQUESTED** [review]

**Pull request closed without merging**